core/src/main/java/com/linecorp/armeria/common/util/SystemInfo.java

```diff
@@ -154,7 +154,10 @@ public static long currentTimeMicros() {
         return JavaVersionSpecific.get().currentTimeMicros();
     }
 
-    private static boolean isLinux() {
+    /**
+     * Return whether Linux is used.
+     */
+    public static boolean isLinux() {
         return Ascii.toLowerCase(System.getProperty("os.name", "")).startsWith("linux");
```

Comment:
Now that this is a public method, we should go ahead and optimize the method. Can you add `private static final boolean IS_LINUX = Ascii.toLowerCase(System.getProperty("os.name", "")).startsWith("linux");` at the beginning and return that here?

Also, anyone have thoughts on making this public means we should instead return an enum of `LINUX`, `WINDOWS`, `MAC_OS_X`, `OTHER`?

Reply:
Yes, how about replacing `isLinux()` with the following method?

```java
private static final OsType OS_TYPE;

static {
    // ... Detect OS type here ...
    OS_TYPE = ...;
}

public static OsType osType() {
    return OS_TYPE;
}
```

`OsType` could look like:

```java
public enum OsType {
    LINUX,
    WINDOWS,
    // MAC_OS_X, FREEBSD, ... (Do we need them?)
    OTHERS
}
```

Reply:
OK 😄 I understand. 
I modified with reference to comment.